### ISSUE TITLE
Path_key not working with multiline core with v1.8.2

### ISSUE THREAD

**Issue by arpantw** [fluent/fluent-bit#3837]
## Bug Report

With multiline core is enabled in fluent-bit v.1.8.2, path_key is not appended to the record. The path_key functionality works fine with the [old multiline parsers](https://docs.fluentbit.io/manual/pipeline/inputs/tail#multiline).

**Steps to reproduce the problem**
- Setup configuration as per https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/multiline-parsing#configuration-example
- Configure input with path_key 
```
[INPUT]
    name             tail
    path             test.log
    read_from_head   true
    multiline.parser multiline-regex-test
    path_key         file_name
```

**Expected behavior**
```
[0] tail.0: [0.000000000, {"log"=>"single line...
","file_name"=>"test.log"}]
```
**Current behavior**
```
[0] tail.0: [0.000000000, {"log"=>"single line...
"}]
```
* Version used: v1.8.2
* Plugins: Tail plugin

**Comment by edsiper** [comment]
troubleshooting this.

workaround: use old multiline mode.

**Comment by edsiper** [comment]
FYI: upcoming solution on PR #3843

**Comment by heitorPB** [comment]
Is this issue fixed and released?

**Comment by kishaningithub** [comment]
Notstale

**Comment by kishaningithub** [comment]
Notstale

**Comment by heitorPB** [comment]
Is this still an issue with 1.8.9?

**Comment by nokute78** [comment]
I confirmed the issue is fixed on v1.8.8.
`"file_name"=>"test.log"` is appended.

```
$ ~/git/WORKTREE/v1.8.8/build/bin/fluent-bit -c a.conf 
Fluent Bit v1.8.8
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/11/08 21:12:58] [ info] [engine] started (pid=20386)
[2021/11/08 21:12:58] [ info] [storage] version=1.1.4, initializing...
[2021/11/08 21:12:58] [ info] [storage] in-memory
[2021/11/08 21:12:58] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/11/08 21:12:58] [ info] [cmetrics] version=0.2.2
[2021/11/08 21:12:58] [ info] [input:tail:tail.0] multiline core started
[2021/11/08 21:12:58] [ info] [sp] stream processor started
[2021/11/08 21:12:58] [ info] [input:tail:tail.0] inotify_fs_add(): inode=1449710 watch_fd=1 name=test.log
[0] tail.0: [1636373578.737956778, {"log"=>"single line...
", "file_name"=>"test.log"}]
[1] tail.0: [1636373578.737968993, {"log"=>"Dec 14 06:41:08 Exception in thread "main" java.lang.RuntimeException: Something has gone wrong, aborting!
    at com.myproject.module.MyProject.badMethod(MyProject.java:22)
    at com.myproject.module.MyProject.oneMoreMethod(MyProject.java:18)
    at com.myproject.module.MyProject.anotherMethod(MyProject.java:14)
    at com.myproject.module.MyProject.someMethod(MyProject.java:10)
    at com.myproject.module.MyProject.main(MyProject.java:6)
", "file_name"=>"test.log"}]
[2] tail.0: [1636373578.737968993, {"log"=>"another line...
", "file_name"=>"test.log"}]
^C[2021/11/08 21:13:01] [engine] caught signal (SIGINT)
[2021/11/08 21:13:01] [ info] [input] pausing tail.0
[2021/11/08 21:13:01] [ warn] [engine] service will stop in 5 seconds
[2021/11/08 21:13:06] [ info] [engine] service stopped
[2021/11/08 21:13:06] [ info] [input:tail:tail.0] inotify_fs_remove(): inode=1449710 watch_fd=1
```

**Comment by arpantw** [comment]
Closing the issue since it has already been fixed.
